### PR TITLE
Implement and use immutable queue for Ripgrep and FilterJob

### PR DESCRIPTION
**Pull request by glennsl**
Performance-wise the only significant difference is that adding items takes twice as long because the length of each chunk is computed twice, inside each of the queues it is added too. Previously it could be computed once since the total count was dealt with separately.

TODO:
- [x] add tests

### CODE REVIEW

File: src/Core/Utility.re [hunk 1]
```diff
@@ -525,3 +525,152 @@ module StringUtil = {
...
+        ([item, ...items], queue);
+
+      | (None, queue) => ([], queue)
+      };
+    };
+
+  let toList = ({front, rear, _}) => front @ List.rev(rear);
+};
+
+module ChunkyQueue: {
```
Comment (bryphe) [inline src/Core/Utility.re]: Very nice @glennsl  - neat abstraction to make the 'chunking' problem simpler

File: test/Core/UtilityTests.re [hunk 2]
```diff
@@ -217,3 +217,256 @@ describe("StringUtil", ({describe, _}) => {
...
+      expect.list(Queue.toList(q)).toEqual([3, 4]);
+    });
+  });
+};
+
+describe("Queue", ({describe, _}) => {
+  testQueue(describe, (module Queue))
+});
+
+describe("ChunkyQueue", ({describe, _}) => {
```
Comment (bryphe) [inline test/Core/UtilityTests.re]: Thanks for the solid set of test cases 💪

**Comment by glennsl** [comment]
I'm pretty sure all CI jobs have succeeded at some point, just not simultaneously, so merging this.

**Pull request merged**